### ISSUE TITLE
used-before-assignment false positive for global variable in function

### ISSUE THREAD

**Issue by fireattack** [PyCQA/pylint#2951]
### Steps to reproduce
```python
def plus_one():
    global count
    count = count + 1

count = 0
plus_one()
```

### Current behavior
```
count: int
Using variable 'count' before assignmentpylint(used-before-assignment)
```
on count = **count** + 1 (bold one)

### Expected behavior

should be no error

### pylint --version output

pylint 2.3.1
astroid 2.2.5
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]

**Comment by AWhetter** [comment]
I think this is a fair message. Because `count` is defined after `plus_one`, it is possible to call `plus_one` before count is defined, and that's a problem. The message is encouraging you to define `count` before defining `plus_one` so that this can never happen.

**Comment by fireattack** [comment]
What do you mean by "possible"? Because `plus_one` is called after count is defined in the example.

**Comment by AWhetter** [comment]
The warning is preventing a mistake being made in a future change to the code. In the future you might change the `plus_one` to happen before `count = 0`, and then you would have a problem. So defining `count` after `plus_one` makes this code more difficult to maintain. Does this make sense?

**Comment by fireattack** [comment]
A warning makes sense, but it shows as an error.
